pkg/auth/oauth/external/handler.go

```diff
@@ -182,3 +183,15 @@ func (defaultState) AuthenticationSucceeded(user user.Info, state string, w http
 	http.Redirect(w, req, then, http.StatusFound)
 	return true, nil
 }
+
+func encodeState(values url.Values) (string, error) {
+	return base64.StdEncoding.EncodeToString([]byte(values.Encode())), nil
```

Comment:
use `base64.URLEncoding` to encode/decode since we're going to use the value in a url